linode/manager
# chore: Remove `url-parse` and clean up query param helpers

## Description 📝

- Remove `url-parse` as a direct dependency because we can use native javascript to do all of this
  - The authors of https://www.npmjs.com/package/url-parse even recommend to **not use this package** in their README.md
- This is basically a follow up of https://github.com/linode/manager/pull/8810

## Major Changes 🔄
- Removes `url-parse` package
- Removes a silly usage of `ramda`
- Updates functions and tests that previously consumed `url-parse`
- Renames functions to they are more clear
- Removes helper functions that had duplicate behavior

## How to test 🧪
- Verify query param related actions are not broken. Some examples include:
  - Table ordering query params
  - Search
  - Crazy deep links like http://localhost:3000/linodes/create?type=One-Click&appID=1102906&utm_source=marketplace&utm_medium=website&utm_campaign=Passky&showInfo=true
